app/src/main/java/org/wikipedia/gallery/GalleryActivity.java

```diff
@@ -534,25 +533,24 @@ private void loadGalleryContent() {
                     getIntent().getStringExtra(EXTRA_FEATURED_IMAGE));
             featuredImage.setAge(getIntent().getIntExtra(EXTRA_FEATURED_IMAGE_AGE, 0));
             updateProgressBar(false);
-            applyGalleryList(Collections.singletonList(featuredImage));
+            applyGalleryList(Collections.singletonList(new MediaListItem(featuredImage.title())));
         } else {
             fetchGalleryItems();
         }
     }
 
-    private void applyGalleryList(@NonNull List<GalleryItem> list) {
+    private void applyGalleryList(@NonNull List<MediaListItem> list) {
         // remove the page transformer while we operate on the pager...
         galleryPager.setPageTransformer(false, null);
         // first, verify that the collection contains the item that the user
         // initially requested, if we have one...
         int initialImagePos = -1;
         if (initialFilename != null) {
-            for (GalleryItem item : list) {
+            for (MediaListItem item : list) {
                 // sometimes the namespace of a file would be in different languages rather than English.
-                String title = StringUtil.removeNamespace(item.getTitles().getCanonical());
-                String titleFromFilePageUrl = StringUtil.removeNamespace(addUnderscores(UriUtil.getTitleFromUrl(item.getFilePage())));
```

Comment:
Not sure how to fix the issue we resolved recently in #517 since there's no `file_page` from the response. The issue will be reproducible now.

Reply:
I'm not able to reproduce the issue. Can you?

Reply:
Although the description and other image information can be displayed successfully, the response from `https://en.wikipedia.org/api/rest_v1/page/media/Badminton_at_the_1992_Summer_Olympics` contains only one file.

If you swipe to the next image, you will see exactly the same image as the previous one.

Reply:
In that case, the `media-list` endpoint should be updated to resolve redirects, and no longer give us redirected titles. We should not be doing the work of matching up redirects.
If this is the only issue remaining, then we can live with it for now.

Reply:
I see. Will create a ticket for that issue.